urllib3/connection.py

```diff
@@ -299,12 +302,26 @@ def connect(self):
                     'for details.)'.format(hostname)),
                     SubjectAltNameWarning
                 )
-            match_hostname(cert, self.assert_hostname or hostname)
+            _match_hostname(cert, self.assert_hostname or hostname)
 
         self.is_verified = (resolved_cert_reqs == ssl.CERT_REQUIRED or
                             self.assert_fingerprint is not None)
 
 
+def _match_hostname(cert, asserted_hostname):
```

Comment:
And I made it private because if there's not some deliberate plan to make it public for some good reason, then better not to expose too much and create something that people rely on and that can't easily be changed in good conscience.